extensions/iceberg/src/main/java/io/deephaven/iceberg/layout/IcebergBaseLayout.java

```diff
@@ -140,6 +146,12 @@ public synchronized void findKeys(@NotNull final Consumer<IcebergTableLocationKe
         try {
             // Retrieve the manifest files from the snapshot
             final List<ManifestFile> manifestFiles = snapshot.allManifests(table.io());
+
+            // Sort manifest files by sequence number to process them in the commit order. This is done as a courtesy
+            // since the actual ordering/sorting of data files is done by the IcebergTableLocationKey implementation.
+            manifestFiles.sort(Comparator.comparingLong(ManifestFile::sequenceNumber));
+            // TODO(deephaven-core#5989): Add unit tests for the ordering of manifest files
+
```

Comment:
I think we should drop this. It likely adds little-to-no benefit (or harm), but it might lead downstream code to be complacent and assume ordering where it should not.
```suggestion
```